src/ape/contracts/base.py

```diff
@@ -463,6 +463,50 @@ def __len__(self):
         logs = self.provider.get_contract_logs(self.log_filter)
         return sum(1 for _ in logs)
 
+    def __call__(self, *args: Any, **kwargs: Any) -> MockContractLog:
+        # Create a dictionary from the positional arguments
+        event_args = dict(zip((input.name for input in self.abi.inputs), args))
+
+        overlapping_keys = set(k for k in event_args.keys() if k is not None) & set(
+            k for k in kwargs.keys() if k is not None
+        )
+
+        if overlapping_keys:
+            raise ValueError(
+                f"Overlapping keys found in arguments: '{', '.join(overlapping_keys)}'."
+            )
+
+        # Update event_args with keyword arguments
+        event_args.update(kwargs)  # type: ignore[arg-type]
+
+        # Check that event_args.keys() is a subset of the expected input names
+        if unknown_input_names := set(event_args.keys()) - {
+            input.name for input in self.abi.inputs
+        }:
+            raise ValueError(
+                f"Invalid argument keys found, expected a subset of {unknown_input_names}"
+            )
+
+        # Convert the arguments using the conversion manager
+        converted_args = {}
+        for key, value in event_args.items():
+            if value is None:
+                continue
+            input_abi = next(input for input in self.abi.inputs if input.name == key)
+            type: Union[str, Type[AddressType], Type[int]] = input_abi.canonical_type
+            if type == "address":
```

Comment:
this might be getting too ecosystem specific here, since we have this same logic in a helper method in the Ethereum class called `self._python_type_for_abi_type`.

Here are some suggestions:

1. Create another `EcosystemAPI` method called `get_python_types(abi_type) -> List[Type]` or something... and share  the logic between the two cases.
2. Create an internal helper method and share the logic and put a TODO comment here to make this more non-ABI specific (non ecosystem specific), meaning less assumptuous  about type names in the ABI

For example, let's say a new zK ecosystem comes out and their ABIs use `contract_id` instead of `address` for whatever reason. In theory, ape should be able to handle that because of the plugin system.

Reply:
I implemented this, where ethereum's implementation calls `self._python_type_for_abi_type` and it seems to be working as intended